prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -301,10 +302,16 @@ void efa_rdm_ope_try_fill_desc(struct efa_rdm_ope *ope, int mr_iov_start, uint64
 				 "buf: %p len: %ld access: %#lx\n",
 				 ope->iov[i].iov_base, ope->iov[i].iov_len, access);
 
-		err = fi_mr_regv(
-			&efa_rdm_ep_domain(ope->ep)->util_domain.domain_fid,
-			ope->iov + i, 1, access, 0, 0, 0, &ope->mr[i],
-			NULL);
+		domain = efa_rdm_ep_domain(ope->ep);
+		if (OFI_LIKELY(efa_mr_cache_enable)) {
```

Comment:
How about we define a function pointer at the domain open for the internal mr_reg call, so you don't have to check this every time

Reply:
Even before my change, `fi_close` is `efa_mr_close` with or without the cache. It always deregistered the MR by calling `ibv_dereg_mr`.

I will fix the other calls to `fi_mr_reg`

Reply:
Talked offline and addressed all comments

I added a function pointer to the domain

We agreed that putting the packet pool MRs in the cache is not useful. I added a comment to reflect that.